CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -123,80 +143,92 @@ function Test-VRouterAgentIntegration {
+
+        Write-Host "===> PASSED: Test-InitialPkt0Injection"
     }
 
     function Test-Pkt0RemainsInjectedAfterAgentStops {
         Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
                [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
 
-        # Given Extension is running
+        Write-Host "===> Running: Test-Pkt0RemainsInjectedAfterAgentStops"
+
+        Write-Host "======> Given Extension is running"
         Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
         Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
         Assert-ExtensionIsRunning -Session $Session -TestConfiguration $TestConfiguration
 
-        # Given Agent is running
+        New-AgentConfigFile -Session $Session -TestConfiguration $TestConfiguration
+
+        Write-Host "======> Given Agent is running"
         Enable-VRouterAgent -Session $Session -ConfigFilePath $TestConfiguration.AgentConfigFilePath
         Assert-AgentIsRunning -Session $Session
         Start-Sleep -Seconds 15
 
-        # Given pkt0 is injected
-        Test-IsPkt0Injected -Session $Session
+        Write-Host "======> Given pkt0 is injected"
+        Assert-IsPkt0Injected -Session $Session
 
-        # When Agent is stopped
+        Write-Host "======> When Agent is stopped"
         Disable-VRouterAgent -Session $Session
         Assert-AgentIsNotRunning -Session $Session
 
-        # Then pkt0 exists in vRouter
+        Write-Host "======> Then pk0 exists in vRouter"
         Assert-IsPkt0Injected -Session $Session
+
+        Write-Host "===> PASSED: Test-Pkt0RemainsInjectedAfterAgentStops"
     }
 
     function Test-OnePkt0ExistsAfterAgentIsRestarted {
         Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
                [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
 
-        # Given Extension is running
+        Write-Host "===> Running: Test-OnePkt0ExistsAfterAgentIsRestarted"
+
+        Write-Host "======> Given Extension is running"
         Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
         Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Assert-ExtensionIsRunning -Session $Session -TestConfiguration $TestConfiguration
 
-        # Given Agent is running
+        New-AgentConfigFile -Session $Session -TestConfiguration $TestConfiguration
+
+        Write-Host "======> Given Agent is running"
         Enable-VRouterAgent -Session $Session -ConfigFilePath $TestConfiguration.AgentConfigFilePath
-        Test-IsVRouterAgentEnable -Session $Session
+        Test-IsVRouterAgentEnabled -Session $Session
         Start-Sleep -Seconds 15
 
-        # Given pkt0 is injected
-        Test-IsPkt0Injected -Session $Session
+        Write-Host "======> Given pkt0 is injected"
```

Comment:
`Start-Sleep -Seconds 15` I think this is unnecessary since we poll agent, right?

Reply:
This `Start-Sleep` is there to ensure that Agent started sending data through KSync pipe. It takes some time for it to start sending data.

I can test for pkt0 in retry loop. What do you think about this?